### PR TITLE
Eth get uncle by block hash and index

### PR DESCRIPTION
**Pull request by Paolob69**
implementation eth_getUncleCountByBlockHash, eth_getUncleByBlockHashAndIndex

### CODE REVIEW

File: silkrpc/json/types.cpp [hunk 1]
```diff
@@ -134,6 +134,7 @@ void to_json(nlohmann::json& json, const Block& b) {
     json["difficulty"] = "0x" + silkrpc::to_hex_no_leading_zeros(silkworm::rlp::big_endian(b.block.header.difficulty));
     json["totalDifficulty"] = "0x" + silkrpc::to_hex_no_leading_zeros(silkworm::rlp::big_endian(b.total_difficulty));
     json["extraData"] = "0x" + silkworm::to_hex(b.block.header.extra_data);
+    json["mixHash"]= b.block.header.mix_hash;
```
Comment (canepat) [inline silkrpc/json/types.cpp:137]: :+1: Kudos for this fix!

File: docs/API.md [hunk 2]
```diff
@@ -21,9 +21,9 @@ The following table shows the current [JSON RPC API](https://eth.wiki/json-rpc/A
 | eth_getBlockByNumber                    | Yes          | not yet tested for performance             |
 | eth_getBlockTransactionCountByHash      | Yes          |                                            |
 | eth_getBlockTransactionCountByNumber    | Yes          |                                            |
-| eth_getUncleByBlockHashAndIndex         | -            | not yet implemented                        |
+| eth_getUncleByBlockHashAndIndex         | Yes          | not yet implemented                        |
```
Comment (canepat) [inline docs/API.md:24]: Remove `not yet implemented`

File: docs/API.md [hunk 3]
```diff
@@ -21,9 +21,9 @@ The following table shows the current [JSON RPC API](https://eth.wiki/json-rpc/A
...
 | eth_getUncleByBlockNumberAndIndex       | -            | not yet implemented                        |
-| eth_getUncleCountByBlockHash            | -            | not yet implemented                        |
+| eth_getUncleCountByBlockHash            | Yes          | not yet implemented                        |
```
Comment (canepat) [inline docs/API.md:26]: Remove `not yet implemented`

File: silkrpc/commands/eth_api.cpp [hunk 4]
```diff
@@ -276,12 +276,41 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_get_block_transaction_count_by_
 
 // https://eth.wiki/json-rpc/API#eth_getunclebyblockhashandindex
 asio::awaitable<void> EthereumRpcApi::handle_eth_get_uncle_by_block_hash_and_index(const nlohmann::json& request, nlohmann::json& reply) {
+    auto params = request["params"];
+    if (params.size() != 2) {
+        auto error_msg = "invalid eth_getUncleByBlockHashIndex params: " + params.dump();
```
Comment (canepat) [inline silkrpc/commands/eth_api.cpp:281]: `eth_getUncleByBlockHashIndex` -> `eth_getUncleByBlockHashAndIndex`

**Review by canepat — CHANGES_REQUESTED** [review]

File: docs/API.md (hunk 2 above)
Comment (Paolob69) [reply docs/API.md:24]: ok

File: docs/API.md (hunk 3 above)
Comment (Paolob69) [reply docs/API.md:26]: ok

File: silkrpc/commands/eth_api.cpp (hunk 4 above)
Comment (Paolob69) [reply silkrpc/commands/eth_api.cpp:281]: ok

**Review by canepat — APPROVED** [review]

**Pull request merged**